ElementX/Sources/Screens/RoomChangePermissionsScreen/View/RoomChangePermissionsScreen.swift

```diff
@@ -14,24 +14,33 @@ struct RoomChangePermissionsScreen: View {
     
     var body: some View {
         Form {
-            ForEach($context.settings) { $setting in
-                Section {
-                    ListRow(label: .plain(title: setting.title),
-                            kind: .inlinePicker(selection: $setting.value, items: setting.allValues))
-                } header: {
-                    Text(setting.title)
-                        .compoundListSectionHeader()
-                }
+            ForEach(RoomChangePermissionsGroup.allCases, id: \.self) { group in
+                section(for: group)
             }
         }
         .compoundList()
-        .navigationTitle(context.viewState.title)
+        .navigationTitle(L10n.screenRoomChangePermissionsTitle)
         .navigationBarTitleDisplayMode(.inline)
         .navigationBarBackButtonHidden(context.viewState.hasChanges)
         .toolbar { toolbar }
         .alert(item: $context.alertInfo)
     }
     
+    @ViewBuilder
+    private func section(for group: RoomChangePermissionsGroup) -> some View {
+        if let settings = $context.settings[group].unwrap() {
```

Comment:
I don't think you need the special `unwrap` extension here, there's an optional init that takes an optional value on `Binding`:

```suggestion
        if let settings = Binding($context.settings[group]) {
```